### PR TITLE
HDDS-9382. Clean up: Remove snapshotId from compaction DAG.

### PR DESCRIPTION
**Pull request by amita-shukla**
## What changes were proposed in this pull request?

As part of initial implementation of RocksDBCheckpointingDiffer, snapshotID was added to [CompactionNode](https://github.com/apache/ozone/pull/3824/files#diff-1d5c6c892a2d24d85ca002dafcb9f39341192baaa7b5323d2325b5fc7a238c7fR246) for debugging. Now in [change#5317](https://github.com/apache/ozone/pull/5317), we migrated to RocksDB to store it and also keep DB transactionId. The other problem is how to get snapshotId. You can argue and use snapshotChainManager but it will create circular dependency unless setter is used (which is not a nice way).

Hence, as part of this PR: 
- the snapshotId from the class `CompactionNode` is removed, including constructor and getter.
- Earlier, in class `RocksDBCheckpointDiffer.java`, the `snapshotId` was being passed as `null`. Removed those occurrences altogether.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9382

## How was this patch tested?

Modified Test class: [TestRocksDBCheckpointDiffer.java](https://github.com/apache/ozone/blob/master/hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdiff/TestRocksDBCheckpointDiffer.java) where the class `CompactionNode` was being used. Ran the test cases from this file to ensure the change does not break any test case.

### CODE REVIEW

File: hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/CompactionNode.java [hunk 1]
```diff
@@ -24,21 +24,18 @@ public class CompactionNode {
   // Name of the SST file
   private final String fileName;
   // The last snapshot created before this node came into existence
```
Comment (hemantk-12) [inline hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/CompactionNode.java:26]: Please remove this comment as well.

**Review by hemantk-12 — COMMENTED** [review]
Thanks for the patch @amita-shukla

Overall looks good to me other than a comment left.

File: hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/CompactionNode.java (hunk 1 above)
Comment (amita-shukla) [reply hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/CompactionNode.java]: Removed, thanks!

File: hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java [hunk 2]
```diff
@@ -541,7 +541,7 @@ public void onCompactionCompleted(RocksDB db,
           // TODO: [SNAPSHOT] Once SnapshotChainManager is put into use,
           //  set snapshotID to snapshotChainManager.getLatestGlobalSnapshot()
```
Comment (hemantk-12) [inline hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java:542]: Can you please remove this comment as well?

**Comment by adoroszlai** [comment]
Thanks @amita-shukla for the patch.  Please enable the `build-branch` workflow in [your fork](https://github.com/amita-shukla/ozone/actions).

File: hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java (hunk 2 above)
Comment (amita-shukla) [reply hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java]: done, thanks!

**Comment by amita-shukla** [comment]
Hi @adoroszlai , the `build-branch` workflow ran for the latest commit but failed: [https://github.com/amita-shukla/ozone/actions/runs/6623599551](https://github.com/amita-shukla/ozone/actions/runs/6623599551). How can this be fixed?

**Comment by adoroszlai** [comment]
> the `build-branch` workflow ran for the latest commit but failed

It failed due to timeout:

```
The operation was canceled.
```

https://github.com/amita-shukla/ozone/actions/runs/6623599551/job/17990948419#step:6:2415

Please select `Re-run failed jobs` (from `Re-run jobs` in the top right).

(Building Ozone requires lots of third-party jars, which are saved in a cache after a successful build.  If the [cache is missing](https://github.com/amita-shukla/ozone/actions/runs/6623599551/job/17990948419#step:4:17) (e.g. on first build), the build takes much longer time.)

**Review by hemantk-12 — APPROVED** [review]
Thanks for the patch @amita-shukla .

Looks good to me.

**Review by swamirishi — APPROVED** [review]
Thanks you for the patch @amita-shukla. Changes look good to me.

**Comment by hemantk-12** [comment]
@amita-shukla Please resolve the conflict and then we can merge it.

**Comment by hemantk-12** [comment]
Thanks for working on this @amita-shukla and @swamirishi for review.

**Pull request merged**